File: register_login_api/app/Models/Question.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Question extends Model
{
    protected $fillable = ['quiz_id', 'question_text', 'correct_answer', 'choices'];
    protected $casts = [
        'choices' => 'array', // Cast 'choices' to array
    ];
    public function quiz()
    {
        return $this->belongsTo(Quiz::class);
    }
}

File: register_login/lib/model/question.dart


class Question {
  final int id;
  final String questionText;
  final String correctAnswer;
  final List<String> choices;

  Question({
    required this.id,
    required this.questionText,
    required this.correctAnswer,
    required this.choices,
  });

  factory Question.fromJson(Map<String, dynamic> json) {
    return Question(
      id: json['id'],
      questionText: json['question_text'],
      correctAnswer: json['correct_answer'],
      choices: List<String>.from(json['choices']),
    );
  }
}
